specification/security/resource-manager/Microsoft.Security/preview/2023-12-01-preview/automations.json

```diff
@@ -0,0 +1,716 @@
+          "description": "The data type of the compared operands (string, integer, floating point number or a boolean [true/false]]",
+          "enum": [
+            "String",
+            "Integer",
+            "Number",
+            "Boolean"
+          ],
+          "x-ms-enum": {
+            "name": "PropertyType",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "String"
+              },
+              {
+                "value": "Integer"
+              },
+              {
+                "value": "Number"
+              },
+              {
+                "value": "Boolean"
+              }
+            ]
+          }
+        },
+        "expectedValue": {
+          "type": "string",
+          "description": "The expected value."
+        },
+        "operator": {
+          "type": "string",
+          "description": "A valid comparer operator to use. A case-insensitive comparison will be applied for String PropertyType.",
+          "enum": [
+            "Equals",
+            "GreaterThan",
+            "GreaterThanOrEqualTo",
+            "LesserThan",
+            "LesserThanOrEqualTo",
+            "NotEquals",
+            "Contains",
+            "StartsWith",
+            "EndsWith"
+          ],
+          "x-ms-enum": {
+            "name": "Operator",
+            "modelAsString": true,
+            "values": [
+              {
+                "value": "Equals",
+                "description": "Applies for decimal and non-decimal operands"
+              },
+              {
+                "value": "GreaterThan",
+                "description": "Applies only for decimal operands"
+              },
+              {
+                "value": "GreaterThanOrEqualTo",
+                "description": "Applies only for decimal operands"
+              },
+              {
+                "value": "LesserThan",
+                "description": "Applies only for decimal operands"
+              },
+              {
+                "value": "LesserThanOrEqualTo",
+                "description": "Applies only for decimal operands"
+              },
+              {
+                "value": "NotEquals",
+                "description": "Applies  for decimal and non-decimal operands"
+              },
+              {
+                "value": "Contains",
+                "description": "Applies only for non-decimal operands"
+              },
+              {
+                "value": "StartsWith",
+                "description": "Applies only for non-decimal operands"
+              },
+              {
+                "value": "EndsWith",
+                "description": "Applies only for non-decimal operands"
+              }
+            ]
+          }
+        }
+      }
+    },
+    "AutomationValidationStatus": {
+      "type": "object",
+      "description": "The security automation model state property bag.",
+      "properties": {
+        "isValid": {
```

Comment:
ARM recommends enums over booleans for future proof APIs.

Standard guidance is: replace boolean/switch properties with a more meaningful enum whenever possible.
A boolean will forever have two valid values (true or false). A string enum type is always preferred. Also, properties should always provide better values just than True and False. For example two switches "isTypeA" and "isTypeB" should be replaced with one enum "type": [A, B, DefaultType]. Enums are always a more flexible and future proof option because they allow additional values to be added in the future in a non-breaking way, e.g. [Enabled, Disabled, Suspended, Deallocated].

Reply:
This property already exists. Changing it now will cause a breaking change, which I would like to avoid. Additionally, as mentioned in the previous comment, I cannot see any additional values that we would want to add. this simply indicates whether it is enabled or not, no other states are applicable here